### PR TITLE
Book store develop

### PR DESCRIPTION
**Pull request by StoneBlood-bit**

### CODE REVIEW

File: src/main/java/mate/academy/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import mate.academy.exception.RegistrationException;
+import mate.academy.service.UserService;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "User registration", description = "Endpoint for registration user")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/auth")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/AuthenticationController.java]:
```suggestion
@RequestMapping("/auth")
```

File: src/main/java/mate/academy/mapper/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class)
+public interface UserMapper {
+    @Mapping(target = "id", source = "id")
+    @Mapping(target = "email", source = "email")
+    @Mapping(target = "firstName", source = "firstName")
+    @Mapping(target = "lastName", source = "lastName")
+    @Mapping(target = "shippingAddress", source = "shippingAddress")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/mapper/UserMapper.java]:
```suggestion
```

File: src/main/java/mate/academy/model/User.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+
+@Entity
+@Table(name = "users")
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/User.java]:
```suggestion
@Getter
@Setter
```

File: src/main/java/mate/academy/service/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/UserServiceImpl.java]:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

File: src/main/java/mate/academy/service/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            throw new RegistrationException("Can't register user");
+        }
+        User user = new User();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/UserServiceImpl.java]: use mapper

File: src/main/java/mate/academy/service/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        user.setEmail(requestDto.getEmail());
+        user.setPassword(requestDto.getPassword());
+        user.setFirstName(requestDto.getFirstName());
+        user.setLastName(requestDto.getLastName());
+        user.setShippingAddress(requestDto.getShippingAddress());
+
+        User savedUser = userRepository.save(user);
+        return userMapper.toUserResponse(savedUser);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/UserServiceImpl.java]:
```suggestion
        userRepository.save(user);
        return userMapper.toUserResponse(user);
```

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                  constraints:
+                    nullable: false
+                    unique: true
+              - column:
+                  name: password
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: firstName
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]:
```suggestion
                  name: first_name
```

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: lastName
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]:
```suggestion
                  name: last_name
```

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: shippingAddress
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]:
```suggestion
                  name: shipping_address
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: last_name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(255)
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:39]: add one empty line

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! 👍

**Pull request merged**